bin/create-docs-pr.js

```diff
@@ -0,0 +1,303 @@
+ * @param {boolean} dryRun skip branch creation
+ */
+async function createBranch(filePath, version, dryRun) {
+  filePath = path.resolve(filePath)
+  console.log(`Changing to ${filePath}`)
+  process.chdir(filePath)
+  const branchName = `add-node-${version}`
+  if (dryRun) {
+    console.log(`Dry run indicated (--dry-run), not creating branch ${branchName}`)
+  } else {
+    console.log('Creating and checking out new branch: ', branchName)
+    await git.checkoutNewBranch(branchName)
+  }
+
+  return branchName
+}
+
+/**
+ * Formats the .mdx to adhere to the docs team standards for
+ * release notes.
+ *
+ * @param {string} releaseDate
+ * @param {string} version
+ * @param {string} body list of changes
+ */
+function formatReleaseNotes(releaseDate, version, body) {
+  const releaseNotesBody = [
+    '---',
+    'subject: Node.js agent',
+    `releaseDate: '${releaseDate}'`,
+    `version: ${version.substr(1)}`,
+    `downloadLink: 'https://www.npmjs.com/package/newrelic'`,
+    '---',
+    '',
+    '##Notes',
+    '',
+    body
+  ].join('\n')
+
+  console.log(`Release Notes Body \n${releaseNotesBody}`)
+  return releaseNotesBody
+}
+
+/**
+ * Writes the contents of the release notes to the docs-website fork
+ *
+ * @param {string} body contents of the .mdx
+ * @param {string} version
+ */
+function addReleaseNotesFile(body, version) {
+  const FILE = getFileName(version)
+  return new Promise((resolve, reject) => {
+    fs.writeFile(FILE, body, 'utf8', (writeErr) => {
+      if (writeErr) {
+        reject(writeErr)
+      }
+
+      console.log(`Added new release notes ${FILE}`)
+      resolve()
+    })
+  })
+}
+
+function getFileName(version) {
+  const FILE = `node-agent-${version}.mdx`
+  return `${RELEASE_NOTES_PATH}/${FILE}`
+}
+
+/**
+ * Commits release notes to the local fork and pushes to proper branch.
+ *
+ * @param {string} version
+ * @param {string} remote
+ * @param {string} branch
+ * @param {boolean} dryRun
+ */
+async function commitRelaseNotes(version, remote, branch, dryRun) {
+  if (dryRun) {
+    console.log('Dry run indicated (--dry-run), skipping committing release notes.')
+    return
+  }
+
+  console.log(`Adding release notes for ${version}`)
+  const files = [getFileName(version)]
+  await git.addFiles(files)
+  await git.commit(`chore: Node.js Agent ${version} Release Notes.`)
+  console.log(`Pushing branch to remote ${remote}`)
+  await git.pushToRemote(remote, branch)
+}
+
+/**
+ * Creates a PR to the newrelic/docs-website with new release notes
+ *
+ * @param {string} username of fork
+ * @param {string} version
+ * @param {string} branch
+ * @param {boolean} dryRun
+ */
+async function createPR(username, version, branch, dryRun) {
+  if (!process.env.GITHUB_TOKEN) {
+    console.log('GITHUB_TOKEN required to create a pull request')
+    stopOnError()
+  }
+
+  const github = new Github('newrelic', 'docs-website')
+  const title = `Node.js Agent ${version} Release Notes`
+  const body = [
```

Comment:
You could prob get rid of most of the formatting here if really wanted. So far, I've typically put minimal messaging (sometimes nothing). https://github.com/newrelic/docs-website/pull/4287